dimod/binary/pybqm.py

```diff
@@ -0,0 +1,336 @@
+        else:
+            return reduce(function, gen, initializer)
+
+    def reduce_neighborhood(self, v: Variable, function: Callable,
+                            initializer: Optional[Any] = None) -> Any:
+        gen = (b for _, b in self.iter_neighborhood(v))
+        if initializer is None:
+            return reduce(function, gen)
+        else:
+            return reduce(function, gen, initializer)
+
+    def reduce_quadratic(self, function: Callable,
+                         initializer: Optional[Any] = None) -> Any:
+        gen = (b for _, _, b in self.iter_quadratic())
+        if initializer is None:
+            return reduce(function, gen)
+        else:
+            return reduce(function, gen, initializer)
+
+    def remove_interaction(self, u: Variable, v: Variable):
+        if u == v:
+            raise ValueError(f"{u!r} cannot have an interaction with itself")
+        try:
+            self._adj[u].pop(v)
+        except KeyError:
+            raise ValueError(f"{u!r} and {v!r} have no interaction") from None
+
+        self._adj[v].pop(u)
+
+    def relabel_variables(self, mapping: Mapping[Variable, Variable]):
+        adj = self._adj
+
+        for submap in iter_safe_relabels(mapping, self.variables):
+            for old, new in submap.items():
+                if old == new:
+                    continue
+
+                # replace the linear bias
+                adj[new] = {new: adj[old].pop(old)}
+
+                # copy the quadratic biases
+                for v in adj[old]:
+                    adj[new][v] = adj[v][new] = adj[v].pop(old)
+
+                # remove the old adj for old
+                del adj[old]
+
+    def relabel_variables_as_integers(self) -> Mapping[int, Variable]:
+        mapping = dict((v, i) for i, v in enumerate(self.variables) if i != v)
+        self.relabel_variables(mapping)
+        return dict((i, v) for v, i in mapping.items())
+
+    def remove_variable(self, v: Optional[Variable] = None) -> Variable:
+        if v is None:
+            try:
+                v, Nv = self._adj.popitem()
+            except KeyError:
+                raise ValueError("cannot pop from an empty model") from None
+        else:
+            try:
+                Nv = self._adj.pop(v)
+            except KeyError:
+                raise ValueError(f"unknown variable {v!r}") from None
+
+        for u in Nv:
+            if u != v:
+                self._adj[u].pop(v)
+
+        return v
+
+    def resize(self, n: int):
+        while n > self.num_variables():
+            self.add_variable()
+        while n < self.num_variables():
+            self.remove_variable()
+
+    def set_linear(self, v: Variable, bias: Any):
+        self._adj.setdefault(v, dict())[v] = bias
+
+    def set_quadratic(self, u: Variable, v: Variable, bias: Any):
+        if u == v:
+            raise ValueError(f"{u!r} cannot have an interaction with itself")
+        self.add_variable(u)
```

Comment:
What if the variables u and v already exists ?

Reply:
`add_variable(u)` does nothing when it already exists.